### PR TITLE
Tests : SearchBar + Api Adresse Services + begin of simulation component

### PR DESCRIPTION
**Pull request by toni-garcia-camptocamp**
Tests : SearchBar + Api Adresse Services + begin of simulation component

### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-328

### CODE REVIEW

File: src/components/simulation/ConsumptionEmailResult.vue [hunk 1]
```diff
@@ -101,8 +99,10 @@ const openPrivacy = () => {
           <span
             @click="openPrivacy()"
             class="font-medium underline decoration-1 hover:cursor-pointer"
-            >Politique de confidentialité</span
+            data-testid="privacy-link"
```
Comment (anaislet-c2c) [inline src/components/simulation/ConsumptionEmailResult.vue:102]: What is the difference between id (l. 75) and data-testid ?

File: src/tests/services/api-rva.spec.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      })
+    )
+
+    const result = await apiRvaService.fetchCommunes(search)
+
+    expect(fetch).toHaveBeenCalledTimes(1)
+    expect(fetch).toHaveBeenCalledWith(
+      `https://api-rva.sig.rennesmetropole.fr/?key=b44535986cf3abf2428d&version=1.0&format=json&epsg=4326&cmd=getcities&query=${search}&insee=all`
+    )
+    expect(result).toEqual([{ name: 'City 1' }, { name: 'City 2' }])
```
Comment (anaislet-c2c) [inline src/tests/services/api-rva.spec.ts:95]: If the url is wrong, does the test return the mockCities values anyway ?

File: src/tests/components/simulation/ConsumptionEmailResult.spec.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { mount } from '@vue/test-utils'
+import ConsumptionEmailResult from '@/components/simulation/ConsumptionEmailResult.vue'
+import CheckBox from '@/components/simulation/CheckBox.vue'
+import { legalList } from '@/constants/legalLinks'
+
+describe('ConsumptionEmailResult', () => {
+  it('renders correctly', () => {
+    const wrapper = mount(ConsumptionEmailResult)
+
+    expect(wrapper.html()).toMatchSnapshot()
```
Comment (anaislet-c2c) [inline src/tests/components/simulation/ConsumptionEmailResult.spec.ts:10]: What is toMatchSnapshot used for ? If the component is modified, does this comparison evolve and/or become false ?

File: src/components/simulation/ConsumptionEmailResult.vue (hunk 1 above)
Comment (toni-garcia-camptocamp) [reply src/components/simulation/ConsumptionEmailResult.vue:102]: data-testid is an id made only for test. Id is the generic id of element

File: src/tests/services/api-rva.spec.ts (hunk 2 above)
Comment (toni-garcia-camptocamp) [reply src/tests/services/api-rva.spec.ts:95]: This allows you to simulate a data format that is returned by the API. In the api-rva file for example, on an isDataValid function that calls answer.status.code:
image

**Pull request merged**